### DISCUSSION TITLE
如何排查大查询负载不均衡问题

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#26283] [Q&A and General discussion]
你好，我现在的 milvus 集群，每个节点的机器配置是 128 核 512GB 的 RAM

一共有 7 个机器，每个机器上安装了一个 querynode

每个机器各一个querynode，一共 7 个 querynode

----

副本设置为 7 个副本、3 个分片

------

然后现在搜索的时候遇到一个问题

就是搜索负载不均衡，有的机器满载，有的机器半载，有的机器零星有几个任务

------

这类问题如何排查呢？

-----

母本大概是 1200w 的向量

-----

样本的话，每次提交查询的样本中包含的向量数在 100-10000 都有，一般都是一次提交上千个向量

然后提交次数不多，可能几分钟一次

查询的不会插入新的向量

-----

针对查询负载不均衡的问题如何排查呢？

**Comment by xiaofan-luan** [comment]
> 

1).取决于你有多少数据，如果数据量足够在7台机器上分，我建议你不要用replica功能。每个查询只能在一个replica使用，并发不够高的情况下可能并不能把所有replica使用起来。
2）如果你的nq很大，但是查询不频繁，可以把nq拆小，这样就能把更多机器利用起来。

**Reply by ponponon** [reply]
1.母本大概是 1200w 的向量,nlist 是 4096
2. 加入我的 nq 是 10000，貌似只有几台机器会工作是吗？我的问题这如果这台机器有有 64 core，这 64 core 会平分  10000个 nq 吗？并且调用 64 core 一起查询搜索吗？还是 1 core 工作，63 coe 围观？
